dissect/target/plugins/os/unix/_os.py

```diff
@@ -149,14 +149,13 @@ def architecture(self) -> str | None:
 
     @export(property=True)
     def hostname(self) -> str | None:
-        hosts_string = self._hosts_dict.get("hostname", "localhost")
-        return self._hostname_dict.get("hostname", hosts_string)
+        return self._hostname or self._hosts_dict.get("hostname", "localhost")
 
     @export(property=True)
     def domain(self) -> str | None:
-        domain = self._hostname_dict.get("domain", "localhost")
+        domain = self._domain or "localhost"
         if domain == "localhost":
-            domain = self._hosts_dict["hostname", "localhost"]
+            domain = self._hosts_dict.get("hostname", "localhost")
```

Comment:
Hah, that looks like a fun bug.

Reply:
Just because we're here anyway, could you add a unit test that would cover this scenario?

Reply:
Very fair. I adjusts the test case in [9394c1c](https://github.com/fox-it/dissect.target/pull/1079/commits/9394c1c875dccea8d7bae334846703e6e949351c) and added some additional test cases just to be sure. Please see if you agree with them.  I also adjusted some other tests to take into account some of the `localhost` behaviour.

Reply:
Ill change the behaviour back later today.